waddrmgr/scoped_manager.go

```diff
@@ -775,9 +775,7 @@ func (s *ScopedKeyManager) deriveKeyFromPath(ns walletdb.ReadBucket,
 func (s *ScopedKeyManager) chainAddressRowToManaged(ns walletdb.ReadBucket,
 	row *dbChainAddressRow) (ManagedAddress, error) {
 
-	// Since the manger's mutex is assumed to held when invoking this
```

Comment:
I feel like using `atomic.Bool` is safer than just not using the mutexes at all but if you disagree, please let me know and I'll change the rest of the code to do it this way. @guggero @Roasbeef

Reply:
yeah I like `atomic.Bool` - unless there is a set of states must be updated together I always prefer using it as it's way cheaper. And I think it makes sense to use it here.